monkey/infection_monkey/system_info/collectors/environment_collector.py

```diff
@@ -0,0 +1,19 @@
+from common.cloud.all_instances import get_all_cloud_instances
+from common.cloud.environment_names import ON_PREMISE
+from infection_monkey.system_info.system_info_collector import SystemInfoCollector
+
+
+def get_monkey_environment() -> str:
+    env = ON_PREMISE
+    for instance in get_all_cloud_instances():
+        if instance.is_instance():
+            env = instance.get_cloud_provider_name()
+    return env
+
+
+class EnvironmentCollector(SystemInfoCollector):
+    def __init__(self):
+        super(EnvironmentCollector, self).__init__(name="EnvironmentCollector")
```

Comment:
Outdated `super` call, `super()` should be enough in py3. You'll probably say explicit is better, but IMO `super()` only means one thing so it's pretty explicit, and whenever we can write less code we should.

Reply:
Oh cool, wasn't aware of the new syntax. 
We have like 80 other uses of old-style matches other than my new ones, so we should probably fix them as well to keep consistency

Reply:
ab330219